src/components/views/rooms/PinnedEventTile.tsx

```diff
@@ -15,112 +15,208 @@ See the License for the specific language governing permissions and
+                    member={event.sender}
+                    size={AVATAR_SIZE}
+                    fallbackUserId={sender}
+                />
+            </div>
+            <div className="mx_PinnedEventTile_wrapper">
+                <div className="mx_PinnedEventTile_top">
+                    <Text
+                        weight="semibold"
+                        className={classNames("mx_PinnedEventTile_sender", getUserNameColorClass(sender))}
+                        as="span"
+                    >
+                        {event.sender?.name || sender}
+                    </Text>
+                    <PinMenu event={event} room={room} permalinkCreator={permalinkCreator} />
+                </div>
+                <MessageEvent
+                    mxEvent={event}
+                    maxImageHeight={150}
+                    onHeightChanged={() => {}} // we need to give this, apparently
+                    permalinkCreator={permalinkCreator}
+                    replacingEventId={event.replacingEventId()}
+                />
+            </div>
+        </div>
+    );
+}
 
-    private onTileClicked = (): void => {
+/**
+ * Properties for {@link PinMenu}.
+ */
+interface PinMenuProps extends PinnedEventTileProps {}
+
+/**
+ * A popover menu with actions on the pinned event
+ */
+function PinMenu({ event, room, permalinkCreator }: PinMenuProps): JSX.Element {
+    const [open, setOpen] = useState(false);
+    const matrixClient = useMatrixClientContext();
+
+    /**
+     * View the event in the timeline.
+     */
+    const onViewInTimeline = useCallback(() => {
         dis.dispatch<ViewRoomPayload>({
             action: Action.ViewRoom,
-            event_id: this.props.event.getId(),
+            event_id: event.getId(),
             highlighted: true,
-            room_id: this.props.event.getRoomId(),
+            room_id: event.getRoomId(),
             metricsTrigger: undefined, // room doesn't change
         });
-    };
-
-    // For event types like polls that use relations, we fetch those manually on
-    // mount and store them here, exposing them through getRelationsForEvent
-    private relations = new Map<string, Map<string, Relations>>();
-    private getRelationsForEvent = (
-        eventId: string,
-        relationType: RelationType | string,
-        eventType: EventType | string,
-    ): Relations | undefined => {
-        if (eventId === this.props.event.getId()) {
-            return this.relations.get(relationType)?.get(eventType);
-        }
-    };
-
-    public render(): React.ReactNode {
-        const sender = this.props.event.getSender();
-
-        if (!sender) {
-            throw new Error("Pinned event unexpectedly has no sender");
+    }, [event]);
+
+    /**
+     * Whether the client can unpin the event.
+     */
+    const canUnpin = useRoomState(room, (state) =>
+        state.mayClientSendStateEvent(EventType.RoomPinnedEvents, matrixClient),
+    );
```

Comment:
Given the permissions is the same for both pinning & unpinning it might be worth clarifying in at least the comment

Reply:
Hmm I don't understand. The permission is only for unpinning here. It's about displaying the unpin button in the contextual menu of the pinned message tile.

Reply:
Sure, but the actual check it does isn't related specifically to unpinning, so a caller may not understand that it is the same permission for pinning & unpinning without having to read the code when considering what underlying Matrix permissions a given user requires to get `true` here. Unpinning sounds like a redaction would be used but that is not the case and thus it maps to the same permission as pinning